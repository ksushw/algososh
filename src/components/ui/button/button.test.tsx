import renderer from "react-test-renderer";
import { Button } from "./button";

it("Кнопка с текстом рендериться без ошибок", () => {
  const tree = renderer.create(<Button text="Рецепт пельменей" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кнопка без текста рендериться без ошибок", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кнопка с индикацией загрузки рендериться без ошибок", () => {
  const tree = renderer.create(<Button isLoader={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Заблокированная кнопка рендериться без ошибок", () => {
  const tree = renderer.create(<Button disabled={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
