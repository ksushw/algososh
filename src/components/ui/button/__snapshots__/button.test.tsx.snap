// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Заблокированная кнопка рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка без текста рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка с индикацией загрузки рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка с текстом рендериться без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Рецепт пельменей
  </p>
</button>
`;
